form.py

```diff
@@ -267,27 +275,27 @@ def review():
                 report.proud_of_accomplishment = proud
                 report.is_final = is_final
                 
-                db.session.add(report)
-
-                try:
-                    sdt = students.query.filter_by(id=getID()).first()
-                    state = sdt.active
-                    if sdt is None:
-                        displayError('user not found in database')
-                    if state == 'midterm':
-                        # check if already submitted
-                        sdt.midterm_done = 1
-                    elif state == 'final':
-                        # check if already submitted
-                        sdt.final_done = 1
-                    else:
-                        displayError('submitting reports not open')
-
-                    db.session.commit()
-                except:
-                    displayError('submission error')
+                gbmodel.db_session.add(report)
+
+        try:
+            #students = gbmodel.students()
+            #sdt = students.query.filter_by(id=getID()).first()
+            sdt = gbmodel.db_session.query(gbmodel.students).filter_by(id=getID()).first()
+            state = sdt.active
```

Comment:
Shouldn't the std.active assignment come after the sdt is None check?

Reply:
fixed